src/types.ts

```diff
@@ -1,64 +1,45 @@
-import type { RequestError } from "@octokit/request-error";
-import type { Schema } from "@octokit/webhooks-definitions/schema";
-import type { EmitterEventWebhookPayloadMap } from "./generated/get-webhook-payload-type-from-event";
-
-type EmitterEventPayloadMap = { "*": Schema } & EmitterEventWebhookPayloadMap;
-
-export type EmitterWebhookEventMap = {
-  [K in keyof EmitterEventPayloadMap]: BaseWebhookEvent<K>;
-};
-
-export type EmitterWebhookEventName = keyof EmitterWebhookEventMap;
-export type EmitterWebhookEvent = EmitterWebhookEventMap[EmitterWebhookEventName];
-
-/**
- * A map of all possible emitter events to their event type.
- * AKA "if the emitter emits x, the handler will be passed y"
- */
-export type EmitterEventMap = EmitterWebhookEventMap & {
-  error: WebhookEventHandlerError;
-};
-export type EmitterEventName = keyof EmitterEventMap;
-export type EmitterEvent = EmitterEventMap[EmitterEventName];
-
-export type EmitterAnyEvent = EmitterWebhookEventMap["*"];
-
-export type ToWebhookEvent<
-  TEmitterEvent extends string
-> = TEmitterEvent extends `${infer TWebhookEvent}.${string}`
-  ? TWebhookEvent
-  : TEmitterEvent;
-
-interface BaseWebhookEvent<
-  TName extends keyof EmitterEventPayloadMap = keyof EmitterEventPayloadMap
-> {
+import { RequestError } from "@octokit/request-error";
+import type {
+  WebhookEventMap,
+  WebhookEventName,
+} from "@octokit/webhooks-definitions/schema";
+import type { emitterEventNames } from "./generated/webhook-names";
+
+export type EmitterWebhookEventName = typeof emitterEventNames[number];
```

Comment:
```suggestion
export type EmitterEventName = typeof emitterEventNames[number];
```
What do you think of just `EmitterEventName`, now every `EmitterEventName` is an `EmitterWebhookEventName` (`"*"` and `"error"` having been removed)?

Reply:
[yup](https://github.com/octokit/webhooks.js/pull/444#issuecomment-774778669), but I'm not so concerned about that since its still accurate with the current name, and everyones approved this version 🤷🙂

Reply:
Following the pattern of `TEmitterEvent` vs. `TWebhookEvent`: https://github.com/octokit/webhooks.js/blob/fc5277d9b315fe5faad8827c8f0716032ce25b0c/src/types.ts#L11